test/modules/generic-mqtt-tester/src/message_initiator.rs

```diff
@@ -57,13 +49,15 @@ impl MessageInitiator {
 
         let mut seq_num: u32 = 0;
         let mut publish_handle = self.publish_handle.clone();
+        let dummy_payload = "a".repeat(self.settings.message_size_in_bytes() as usize);
         loop {
             info!("publishing message {} to upstream broker", seq_num);
+
             let publication = Publication {
-                topic_name: FORWARDS_TOPIC.to_string(),
+                topic_name: self.settings.initiate_topic(),
                 qos: QoS::ExactlyOnce,
                 retain: true,
-                payload: Bytes::from(seq_num.to_string()),
+                payload: Bytes::from(seq_num.to_string() + " " + dummy_payload.as_str()),
```

Comment:
Reading seq num part looks good, but I think there is a problem here, since you're writing seq_num as a string, and reading it as bytes. 

After playing a bit with it, I think it should look similar to below example. 

It looks like `Bytes::from(&str)` also does an allocation anyways, so in the example below we still have just one allocation `BytesMut::with_capacity()`. Then we write seq num as bytes and the dummy data into that buffer. `payload.into()` just converts `BytesMut` into `Bytes`, no extra allocations, afaik.

I have not tested it, just checked that it compiles :)

```rust
        let payload_size = self.settings.message_size_in_bytes() as usize;
        let dummy_data = &vec!['a' as u8; payload_size];
        loop {
            info!("publishing message {} to upstream broker", seq_num);

            let mut payload = BytesMut::with_capacity(payload_size + 4);
            payload.put_u32(seq_num);
            payload.put_slice(&dummy_data);
            let publication = Publication {
                topic_name: self.settings.initiate_topic(),
                qos: QoS::ExactlyOnce,
                retain: true,
                payload: payload.into(),
            };
```